tests/unit/inference/quantization/test_intX_quantization.py

```diff
@@ -326,9 +329,9 @@ def test_quantized_linear(self):
                 'weight_quantization': {
                     'post_init_quant': {
                         'layer': {
-                            'num_bits': 4,
+                            'num_bits': quantization_bits,
                             'group_size': 64,
-                            'group_dim': 0,
+                            'group_dim': 1,
```

Comment:
Why this change?

Reply:
nice catch. fixed it.

Reply:
we need to cover group_dim along the last dimension in the tensor to invoke the cuda dequantization kernels. otherwise, it will call reference torch version([link](https://github.com/microsoft/DeepSpeed/blob/0b21637881b562ab0ca0ab9caadce0555d221912/deepspeed/inference/quantization/utils.py#L109C1-L109C72)). so i changed the code to cover both cases with group_dim 0 and 1 from this test.